php/libraries/NDB_Page.class.inc

```diff
@@ -187,7 +187,7 @@ class NDB_Page implements RequestHandlerInterface
      *
      * @return void
      */
-    function addSelect($name, $label, $options, $attribs=array())
+    function addSelect(string $name, string $label, array $options, array $attribs=array())
```

Comment:
Looks good. Could you types to the arrays in the comment if known? e.g. changing line 183 to

      * @param array<string,string>  $options Options to pass to QuickForm for this

Reply:
I don't know how it's being called, I just based the php types on the phpdoc and know that the places where `$attribs` was a string are wrong..

Reply:
@johnsaigle does php allow type enforcement within arrays now?

Reply:
Not exactly. Some tools know how to analyze the PHPDocs and do additional validation if you specify that all entries need to be a specific type but something else gets passed in. It's not built into PHP itself.

Reply:
@HenriRabalais No, he's referring to the doc comments. phan does analysis based on them in addition to the PHP types